### ISSUE TITLE
"Error The type 'System.Xml.Serialization.IXmlSerializable' is defined in an assembly that is not referenced."

### ISSUE THREAD

**Issue by dahall** [dahall/TaskScheduler#128]
Hello,

I am not sure if I should be posting this under issue, but it seems right.

I would like to use your wrapper for a project of mine, but something weird is happening when I try to run it. I am using the latest version (today is 2013/06/18, using the Release 1.9.4).

When I run this piece of code:

```
using (TaskService ts = new TaskService())
            {
                TimeTrigger tt = new TimeTrigger { StartBoundary = new DateTime(2013, 6, 18, 19, 35, 00) };
                ts.AddTask("TaskName",
                   tt,
                   new ExecAction("notepad.exe", null, null));
                
            }
```
it works perfectly fine. But if I change TimeTrigger to DailyTrigger, and nothing else, I get a compile error.

The exact error message is "Error	1	The type 'System.Xml.Serialization.IXmlSerializable' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.Xml, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'."

Please let me know what's the reason for this.
I am using Visual Studio 2012 Ultimate, and I created an empty project, with one Program.cs file in whose main I call this piece of code.

Best regards!

Originally posted: 2013-06-18T11:46:44

**Comment by dahall** [comment]
Will you post the code for how you are changing the TimeTrigger to the DailyTrigger? This may be why the error is occurring. Typically, you would need to retrieve the task, remove the TimeTrigger, add the DailyTrigger and then re-register the task. Is this what you are doing?

Originally posted: 2013-06-18T12:10:27

**Comment by dahall** [comment]
That probably might be why it's not working.
Like I said, in the same project, same .cs file, I just change the TimeTrigger (in the same TaskService) to DailyTrigger, like this: 

```
using (TaskService ts = new TaskService())
            {
                DailyTrigger tt = new DailyTrigger { StartBoundary = new DateTime(2013, 6, 18, 19, 35, 00) };
                ts.AddTask("TaskName",
                   tt,
                   new ExecAction("notepad.exe", null, null));
                
            }
```

Is there somewhere an example how to do what you said?


Originally posted: 2013-06-19T04:04:16

**Comment by dahall** [comment]
Each time you need to change a task, after it is created, the following code would work:
```
// Retrieve the task, change the trigger and re-register it
Task t = ts.GetTask("TaskName");
td = t.Definition;
td.Triggers.Clear();
td.Triggers.Add(new DailyTrigger { StartBoundary = new DateTime(2013, 6, 18, 19, 35, 00) });
t.RegisterChanges();
```

Originally posted: 2013-06-19T07:21:21

**Comment by dahall** [comment]
I pasted this piece of code in my project, I even tried it in a new, different project, and I still get the same compile error.

The only thing I changed/added is on the second line of your code:
```
td = t.Definition;
```

changed to
```
TaskDefinition td = t.Definition;
```
(because td isn't initialized).

And VS underlines the TaskDefinition part of that particular line.


Am I really the only one who has this problem?

Originally posted: 2013-06-21T08:05:22
